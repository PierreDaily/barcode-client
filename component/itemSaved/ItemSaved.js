import React from "react";
import { View, Text } from "react-native";
import { Button } from "react-native-paper";

export default ({ navigation }) => (
  <View>
    <Text>Thank you for your help, your item has been submited</Text>
    <Button mode="contained" onPress={() => navigation.navigate("Home")}>Home</Button>
  </View>
);
