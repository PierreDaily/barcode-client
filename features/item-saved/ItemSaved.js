import React from "react";
import PropTypes from "prop-types";
import { View, Text } from "react-native";
import { Button } from "react-native-paper";

const ItemSaved = ({ navigation }) => (
  <View>
    <Text>Thank you for your help, your item has been submited</Text>
    <Button mode="contained" onPress={() => navigation.navigate("Home")}>
      Home
    </Button>
  </View>
);

ItemSaved.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  })
};

export { ItemSaved };
