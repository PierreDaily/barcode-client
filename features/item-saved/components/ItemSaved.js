import React from "react";
import PropTypes from "prop-types";
import { View, Text } from "react-native";
import { Button } from "../../ui";
import styles from "../styles";
import CheckCircle from "../../../assets/img/green-check-circle.svg";

const ItemSaved = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <CheckCircle style={styles.checkCircle} />
        <Text style={styles.text}>Your item has been submited.</Text>
        <Text style={styles.text}>
          Would you like to help the community and input the price ?
        </Text>
      </View>
      <Button
        style={styles.btn__yes}
        onPress={() => {
          navigation.navigate("Add-price");
        }}
        testID="yesBtn"
      >
        Yes
      </Button>
      <Button
        style={styles.btn}
        onPress={() => navigation.navigate("Home")}
        testID="noBtn"
      >
        No
      </Button>
    </View>
  );
};

ItemSaved.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  })
};

export { ItemSaved };
