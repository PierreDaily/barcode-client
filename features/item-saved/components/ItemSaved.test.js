import React from "react";
import { render, fireEvent } from "react-native-testing-library";
import { ItemSaved } from "./ItemSaved";

import renderer from "react-test-renderer";

let navigationMock;

jest.mock("../../../assets/img/green-check-circle.svg", () => ({
  __esModule: true,
  default: "CheckCircle"
}));

afterEach(() => {
  jest.clearAllMocks();
});

beforeEach(() => {
  navigationMock = { navigate: jest.fn() };
});

test("renders correctly", () => {
  const tree = renderer.create(<ItemSaved />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("navigates to Home screen on 'No' button press", () => {
  const { getByTestId } = render(<ItemSaved navigation={navigationMock} />);

  const noButton = getByTestId("noBtn");

  fireEvent.press(noButton);
  expect(navigationMock.navigate).toHaveBeenCalledWith("Home");
});

test("navigates to Add-price screen on 'Yes' button press", () => {
  const { getByTestId } = render(<ItemSaved navigation={navigationMock} />);

  const yesButton = getByTestId("yesBtn");

  fireEvent.press(yesButton);
  expect(navigationMock.navigate).toHaveBeenCalledWith("Add-price");
});
