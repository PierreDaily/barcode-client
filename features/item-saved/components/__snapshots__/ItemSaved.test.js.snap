// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 40,
      "paddingRight": 40,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 260,
        "top": 133,
      }
    }
  >
    <CheckCircle
      style={
        Object {
          "marginBottom": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#4E66A9",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Your item has been submited.
    </Text>
    <Text
      style={
        Object {
          "color": "#4E66A9",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Would you like to help the community and input the price ?
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 0,
        "bottom": 92,
        "elevation": 2,
        "height": 40,
        "marginLeft": 40,
        "minWidth": 64,
        "position": "absolute",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
    testID="yesBtn"
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#F8F8F8",
                },
              ],
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 0,
        "bottom": 46,
        "elevation": 2,
        "height": 40,
        "marginLeft": 40,
        "minWidth": 64,
        "position": "absolute",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
    testID="noBtn"
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "#F8F8F8",
                },
              ],
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</View>
`;
